File: Objective-c/pamservices/FingerPrint.m
//
//  FingerPrint.m
//  pamservices
//
//  Created by heart on 9/8/2560 BE.
//  Copyright © 2560 heart. All rights reserved.
//

#import "FingerPrint.h"
#import <CommonCrypto/CommonDigest.h>

@implementation FingerPrint

+(NSString*)fromString:(NSString*)string{
    NSData *data = [string dataUsingEncoding:NSUTF8StringEncoding];
    uint8_t digest[CC_SHA1_DIGEST_LENGTH];
    CC_SHA1(data.bytes, (CC_LONG)data.length, digest);
    NSMutableString *output = [NSMutableString stringWithCapacity:CC_SHA1_DIGEST_LENGTH * 2];
    
    for (int i = 0; i < CC_SHA1_DIGEST_LENGTH; i++)
    {
        [output appendFormat:@"%02x", digest[i]];
    }
    
    return output;
}

@end

File: Swift/pamservices/FingerPrint.swift
//
//  FingerPrint.swift
//  pamservices
//
//  Created by narongrit kanhanoi on 9/8/2560 BE.
//  Copyright © 2017 pushandmotion.com. All rights reserved.
//

class FingerPrint {
    class func from(_ string: String) -> String? {
        return SHA1.hexString(from: string)
    }
}
